### PR TITLE
UI & branding improvements, notifications, manual dark mode, and more message types

### LINKED ISSUE #26: Force dark mode toggle not working
There is a "force dark mode" toggle button in the settings tab that should allow web users and those without an Android 10+ device to enable Itchy's dark theme via a CSS class.  However, I haven't been able to get this to work.  Help is appreciated!

### PR DESCRIPTION
**Pull request by micahlt**
This is possibly the biggest block of changes since Itchy moved from Cordova to Capacitor.  I've tried to organize my commits as much as I can, so it should be relatively easy to review.  Here is a zipped APK file for testing: [0-9-1-prerelease.zip](https://github.com/Scratch-Client-4/itchy-ionic/files/7696099/0-9-1-prerelease.zip)

**Breaking changes:** 

- Upgraded to Android Studio Arctic Fox (latest)
- Upgraded to Ionic 5
- Upgraded to Capacitor 3.x
- Updated to Gradle 7

**User-facing changes:**

- More supported message types
- More supported feed types
- New (non-copyright-infringing) branding
- Share button for projects
- Local notifications every 30 minutes
- "Projects Friends Loved" section on Explore
- Much better user page
- Dynamic status bar colors to match dark mode

**Backend changes:**

- Breaking changes listed above
- Addition of `Share`, `App`, and `LocalNotifications` Capacitor plugins
- More requests done locally rather than relying on `itchy-api`
- Fixed back button behavior in modals
- Fixed manual dark mode toggle (close #26)
- Moved to ScratchDB v3 by DatOneLefty

### CODE REVIEW

**Comment by locness3** [comment]
> "Projects Friends Loved" section on Explore

Does that mean followed people?

> Much better user page

And broken "Open your profile" link in Settings.
Also, opening your profile from the settings, then closing it, then going to Explore will open it again

> Dynamic status bar colors to match dark mode

Both my status bar and navigation bar are black on light mode. Is this intended? (android 9)

File: src/components/Message.vue [hunk 1]
```diff
@@ -27,15 +27,21 @@
     <ion-note v-if="m.type == 'remixproject'">
       <ion-icon :icon="colorPalette" class="blue"></ion-icon> remixed your project {{ m.parent_title }}
     </ion-note>
-    <ion-note v-if="m.type == 'remixproject'">
-      <ion-icon :icon="colorPalette" class="blue"></ion-icon> remixed your project {{ m.parent_title }}
+    <ion-note v-if="m.type == 'curatorinvite'">
+      <ion-icon :icon="images" class="blue"></ion-icon> wants you to curate {{ m.title }}
```
Comment (locness3) [inline src/components/Message.vue:31]: *invited you

File: src/components/Message.vue [hunk 2]
```diff
@@ -231,34 +290,29 @@ export default defineComponent({
...
       }
       const res = await Http.request(reqOpts);
       if (res.status == 200) {
-        this.toastNotif(`Followed user ${o.actor_username}`);
+        this.toastNotif(`You are a curator of ${o.title}`);
       } else {
         console.error(res.statusText);
       }
     },
-    async thankUser(o) {
```
Comment (locness3) [inline src/components/Message.vue:251]: thank??

File: src/components/ProjectModal.vue [hunk 3]
```diff
@@ -169,10 +184,21 @@ export default defineComponent({
     this.loadProject();
   },
   methods: {
+    async shareProject() {
+      await Share.share({
+        title: this.title,
+        text: 'Check out this Scratch project I found with Itchy!',
```
Comment (locness3) [inline src/components/ProjectModal.vue:190]: IMO, this text is pretty unnecessary. I personnally dislike when apps add text to shared links.

File: src/components/Message.vue (hunk 2 above)
Comment (micahlt) [reply src/components/Message.vue:251]: Essentially this leads the current user to the following user's profile comments to thank them for the follow.

File: src/views/SettingsPage.vue [hunk 4]
```diff
@@ -43,7 +43,7 @@
       </ion-item>
       <ion-item>
         <ion-label>
-          Enable personal feed
+          Enable personal data on Explore
```
Comment (locness3) [inline src/views/SettingsPage.vue:46]: Enable personal feed on Explore.

**Comment by micahlt** [comment]
> > "Projects Friends Loved" section on Explore
> 
> Does that mean followed people?
> 
> > Much better user page
> 
> And broken "Open your profile" link in Settings.
> Also, opening your profile from the settings, then closing it, then going to Explore will open it again
> 
> > Dynamic status bar colors to match dark mode
> 
> Both my status bar and navigation bar are black on light mode. Is this intended? (android 9)


Yes, "Friends" refers to followed users, just like the Scratch 1.x era.  I can fix the "open your profile" link this afternoon.  As far as routing issues, that was an existing issue that will be resolved in a future PR.  What color is your light mode status bar normally?  And imo, does it look bad to you?

File: src/components/Message.vue (hunk 1 above)
Comment (micahlt) [reply src/components/Message.vue:31]: What is the reason for this change?

File: src/components/ProjectModal.vue (hunk 3 above)
Comment (micahlt) [reply src/components/ProjectModal.vue:190]: Agreed.  I'll fix that this afternoon.

**Comment by locness3** [comment]
> What color is your light mode status bar normally?

It always was black AFAIRC, but I assume it should be white on light mode?

> Yes, "Friends" refers to followed users, just like the Scratch 1.x era.
I don't think people who would use this app would be aware - should change this.

File: src/components/Message.vue (hunk 2 above)
Comment (locness3) [reply src/components/Message.vue:251]: I get it - but the function name is not correct here.

File: src/components/Message.vue (hunk 1 above)
Comment (locness3) [reply src/components/Message.vue:31]: Makes it consistent with the website.

**Comment by locness3** [comment]
Screenshot_20211211-151329
Also, I don't have this much messages.

File: src/views/SettingsPage.vue (hunk 4 above)
Comment (micahlt) [reply src/views/SettingsPage.vue:46]: > Enable personal feed on Explore

I would have used this, but it also uses personal data for the "Projects Friends Loved" section.

File: src/components/Message.vue (hunk 2 above)
Comment (micahlt) [reply src/components/Message.vue:251]: What would you like me to rename it?

File: src/components/Message.vue (hunk 1 above)
Comment (micahlt) [reply src/components/Message.vue:31]: Fair enough.  Will fix.

**Comment by micahlt** [comment]
> > What color is your light mode status bar normally?
> 
> It always was black AFAIRC, but I assume it should be white on light mode?
> 
> > Yes, "Friends" refers to followed users, just like the Scratch 1.x era.
> 
> I don't think people who would use this app would be aware - should change this.

Most of the apps I use continue to use a dark status bar even in light mode - I don't think that's an issue.  As far as renaming that section of Explore, I'm not sure.  I think it should be pretty logical since it's already the same layout of the Scratch homepage.  The main reason I used the current phrase was to keep the line length short.  Anything longer than three mid-size words creates overflow which is unacceptable.  Any suggestions?

**Comment by micahlt** [comment]
> Screenshot_20211211-151329
> Also, I don't have this much messages.

Very odd - can you debug with chrome devtools?

**Comment by micahlt** [comment]
I just pushed a fix for the function names and strings (a6987d5).  The messages issue is very odd because I've been using this version of Itchy for around 6 hours and found no issues with the notifications function.  I can't do anything to debug it on my end, so you'll have to take care of it.

**Comment by micahlt** [comment]
With everything working at this point, I'm going to go ahead and merge this.  Since the initial PR, I also: 

- Fixed some navigation issues
- Added more feed events
- Drastically improved profile UI
- Fixed infinite scroll on search
- Improved modularization
- All-around performance improvements

**Pull request merged**